notebooks/text_examples/text_entailment/text_entailment_demo.ipynb

```diff
@@ -0,0 +1,784 @@
+    "true_label = snli_label_map[label]\n",
+    "print('The true label is: {true_label}'.format(true_label=true_label))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "id": "concerned-sequence",
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "contradiction probability: 99.95%\n",
+      "neutral probability: 0.03%\n",
+      "entailment probability: 0.02%\n"
+     ]
+    }
+   ],
+   "source": [
+    "# test model\n",
+    "input_ids = tokenizer.encode(premise, hypothesis, return_tensors='pt')\n",
+    "logits = model(input_ids)[0]\n",
+    "probs = logits.softmax(dim=1)\n",
+    "\n",
+    "bart_label_map = {0: 'contradiction', 1: 'neutral', 2: 'entailment'}\n",
+    "for i, lab in bart_label_map.items():\n",
+    "    print('{lab} probability: {prob:0.2f}%'.format(lab=lab, prob=probs[0][i] * 100))\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "ready-majority",
+   "metadata": {},
+   "source": [
+    "## Run shap values"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "id": "sunrise-modification",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import scipy as sp\n",
+    "import torch\n",
+    "\n",
+    "# wrapper function for model\n",
+    "# takes in masked string which is in the form: premise <separator token(s)> hypothesis \n",
+    "def f(x): \n",
+    "    outputs = []\n",
+    "    for _x in x:\n",
+    "        encoding = torch.tensor([tokenizer.encode(_x)])\n",
+    "        output = model(encoding)[0].detach().cpu().numpy() \n",
+    "        outputs.append(output[0])\n",
+    "    outputs = np.array(outputs)\n",
+    "    scores = (np.exp(outputs).T / np.exp(outputs).sum(-1)).T\n",
+    "    val = sp.special.logit(scores)\n",
+    "    return val"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "id": "realistic-survivor",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "explainers.Partition is still in an alpha state, so use with caution...\n"
+     ]
+    }
+   ],
+   "source": [
+    "# Construct explainer\n",
+    "bart_labels = ['contradiction', 'entailment', 'neutral']    \n",
+    "explainer = shap.Explainer(f, tokenizer, output_names=bart_labels)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "id": "cardiovascular-deposit",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "A boy is jumping on skateboard in the middle of a red bridge.</s></s>The boy skates down the sidewalk.\n"
+     ]
+    }
+   ],
+   "source": [
+    "# encode then decode premise, hypothesis to get concatenated sentences\n",
+    "encoded = tokenizer(premise, hypothesis)['input_ids'][1:-1] # ignore the start and end tokens, since tokenizer will naturally add them\n",
```

Comment:
Hm, is this why there was a difference between your code and Max's code? You ignore the tokens here?